corehq/ex-submodules/couchforms/management/commands/purge_forms_and_cases.py

```diff
@@ -78,6 +82,13 @@ def iterate_forms_and_collect_case_ids(self):
         if self.case_ids:
             self.print_case_details()
 
+    def get_xform_build_version(self, xform):
+        version_from_mapping = self.version_mapping.get(xform.build_id)
+        if not version_from_mapping:
+            version_from_mapping = int(get_app(self.domain, xform.build_id).version)
```

Comment:
Added a check for build_id and then also in case app is not found. Should cover possible edge cases though not tested.